### ISSUE TITLE
Linking Fails in FreeBSD 10.3

### ISSUE THREAD

**Issue by pnovotnak** [fluent/fluent-bit#158]
Maybe I'm making a n00b mistake, but it looks like neither `0.10.0` or `master` will compile on FreeBSD, regardless of version;

```
root@FreeBSD-10:~/fluent-bit/build # cmake ..
-- The C compiler identification is Clang 3.4.1
-- The CXX compiler identification is Clang 3.4.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/CC
-- Check for working CXX compiler: /usr/bin/CC -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find GTest (missing:  GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.8")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Performing Test MSGPACK_ENABLE_GCC_CXX_ATOMIC
-- Performing Test MSGPACK_ENABLE_GCC_CXX_ATOMIC - Failed
-- Performing Test FLB_HAVE_SYNC_SUB
-- Performing Test FLB_HAVE_SYNC_SUB - Success
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE)
-- Performing Test HAVE_STAT_H
-- Performing Test HAVE_STAT_H - Success
-- Performing Test HAVE_SYS_UIO_H
-- Performing Test HAVE_SYS_UIO_H - Success
-- Performing Test HAVE_UNISTD_H
-- Performing Test HAVE_UNISTD_H - Success
-- Performing Test HAVE_EPOLL
-- Performing Test HAVE_EPOLL - Failed
-- Performing Test HAVE_KQUEUE
-- Performing Test HAVE_KQUEUE - Failed
-- Event loop backend > select(2)
-- Performing Test HAVE_TIMERFD_CREATE
-- Performing Test HAVE_TIMERFD_CREATE - Failed
-- Performing Test HAVE_EVENTFD
-- Performing Test HAVE_EVENTFD - Failed
-- Found Perl: /usr/local/bin/perl (found version "5.20.3")
Can't open perl script "/root/fluent-bit/scripts/config.pl": No such file or directory
-- Performing Test FLB_HAVE_UCONTEXT
-- Performing Test FLB_HAVE_UCONTEXT - Success
-- Performing Test FLB_HAVE_VALGRIND
-- Performing Test FLB_HAVE_VALGRIND - Failed
-- Performing Test FLB_HAVE_FORK
-- Performing Test FLB_HAVE_FORK - Success
-- Performing Test FLB_HAVE_C_TLS
-- Performing Test FLB_HAVE_C_TLS - Success
-- Performing Test FLB_HAVE_SETJMP
-- Performing Test FLB_HAVE_SETJMP - Success
-- Performing Test FLB_HAVE_ACCEPT4
-- Performing Test FLB_HAVE_ACCEPT4 - Success
-- Performing Test FLB_HAVE_INOTIFY
-- Performing Test FLB_HAVE_INOTIFY - Failed
-- Looking for accept4
-- Looking for accept4 - found
-- Configuring done
CMake Warning (dev) at src/CMakeLists.txt:153 (add_library):
  Policy CMP0063 is not set: Honor visibility properties for all target
  types.  Run "cmake --help-policy CMP0063" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "fluent-bit-static" of type "STATIC_LIBRARY" has the following
  visibility properties set for C:

    C_VISIBILITY_PRESET

  For compatibility CMake is not honoring them for this target.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/proxy/go/CMakeLists.txt:4 (add_library):
  Policy CMP0063 is not set: Honor visibility properties for all target
  types.  Run "cmake --help-policy CMP0063" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "flb-plugin-proxy-go" of type "STATIC_LIBRARY" has the following
  visibility properties set for C:

    C_VISIBILITY_PRESET

  For compatibility CMake is not honoring them for this target.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /root/fluent-bit/build
root@FreeBSD-10:~/fluent-bit/build # make
Scanning dependencies of target msgpackc-static
[  0%] Building C object lib/msgpack-c-0b7cabd/CMakeFiles/msgpackc-static.dir/src/objectc.c.o
[  1%] Building C object lib/msgpack-c-0b7cabd/CMakeFiles/msgpackc-static.dir/src/unpack.c.o
[  1%] Building C object lib/msgpack-c-0b7cabd/CMakeFiles/msgpackc-static.dir/src/version.c.o
[  2%] Building C object lib/msgpack-c-0b7cabd/CMakeFiles/msgpackc-static.dir/src/vrefbuffer.c.o
[  2%] Building C object lib/msgpack-c-0b7cabd/CMakeFiles/msgpackc-static.dir/src/zone.c.o
[  3%] Linking C static library ../../library/libmsgpackc.a
[  3%] Built target msgpackc-static
Scanning dependencies of target jsmn
[  4%] Building C object lib/jsmn/CMakeFiles/jsmn.dir/jsmn.c.o
[  4%] Linking C static library ../../library/libjsmn.a
[  4%] Built target jsmn
Scanning dependencies of target ejson
[  5%] Building C object lib/ejson/CMakeFiles/ejson.dir/ejson.c.o
/root/fluent-bit/lib/ejson/ejson.c:167:20: warning: passing 'unsigned char *' to parameter of type 'char *' converts between pointers to integer types with different sign [-Wpointer-sign]
    ret = snprintf(buf + ctx->buf_pos,
                   ^~~~~~~~~~~~~~~~~~
/usr/include/stdio.h:282:32: note: passing argument to parameter here
int      snprintf(char * __restrict, size_t, const char * __restrict,
                                   ^
1 warning generated.
[  5%] Linking C static library ../../library/libejson.a
[  5%] Built target ejson
Scanning dependencies of target mk_core
[  6%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_rconf.c.o
[  6%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_string.c.o
[  7%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_memory.c.o
[  7%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_event.c.o
[  8%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_utils.c.o
[  8%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_file.c.o
[  8%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_iov.c.o
[  9%] Linking C static library ../../../library/libmk_core.a
[  9%] Built target mk_core
Scanning dependencies of target mbedcrypto
[  9%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/aes.c.o
[ 10%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/aesni.c.o
[ 10%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/arc4.c.o
[ 11%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/asn1parse.c.o
[ 11%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/asn1write.c.o
[ 12%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/base64.c.o
[ 12%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/bignum.c.o
[ 13%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/blowfish.c.o
[ 13%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/camellia.c.o
[ 14%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ccm.c.o
[ 14%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/cipher.c.o
[ 15%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/cipher_wrap.c.o
[ 15%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/cmac.c.o
[ 15%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ctr_drbg.c.o
[ 16%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/des.c.o
[ 16%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/dhm.c.o
[ 17%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ecdh.c.o
[ 17%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ecdsa.c.o
[ 18%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ecjpake.c.o
[ 18%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ecp.c.o
[ 19%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ecp_curves.c.o
[ 19%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/entropy.c.o
[ 20%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/entropy_poll.c.o
[ 20%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/error.c.o
[ 21%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/gcm.c.o
[ 21%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/havege.c.o
[ 21%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/hmac_drbg.c.o
[ 22%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/md.c.o
[ 22%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/md2.c.o
[ 23%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/md4.c.o
[ 23%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/md5.c.o
[ 24%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/md_wrap.c.o
[ 24%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/memory_buffer_alloc.c.o
[ 25%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/oid.c.o
[ 25%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/padlock.c.o
[ 26%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pem.c.o
[ 26%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pk.c.o
[ 27%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pk_wrap.c.o
[ 27%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pkcs12.c.o
[ 28%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pkcs5.c.o
[ 28%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pkparse.c.o
[ 28%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/pkwrite.c.o
[ 29%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/platform.c.o
[ 29%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/ripemd160.c.o
[ 30%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/rsa.c.o
[ 30%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/sha1.c.o
[ 31%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/sha256.c.o
[ 31%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/sha512.c.o
[ 32%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/threading.c.o
[ 32%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/timing.c.o
[ 33%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/version.c.o
[ 33%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/version_features.c.o
[ 34%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedcrypto.dir/xtea.c.o
[ 34%] Linking C static library ../../../library/libmbedcrypto.a
[ 34%] Built target mbedcrypto
Scanning dependencies of target mbedx509
[ 34%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/certs.c.o
[ 35%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/pkcs11.c.o
[ 35%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509.c.o
[ 35%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509_create.c.o
[ 36%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509_crl.c.o
[ 36%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509_crt.c.o
[ 37%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509_csr.c.o
[ 37%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509write_crt.c.o
[ 38%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedx509.dir/x509write_csr.c.o
[ 38%] Linking C static library ../../../library/libmbedx509.a
[ 38%] Built target mbedx509
Scanning dependencies of target mbedtls
[ 38%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/debug.c.o
[ 39%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/net_sockets.c.o
[ 39%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_cache.c.o
[ 40%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_ciphersuites.c.o
[ 40%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_cli.c.o
[ 41%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_cookie.c.o
[ 41%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_srv.c.o
[ 42%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_ticket.c.o
[ 42%] Building C object lib/mbedtls-2.4.1/library/CMakeFiles/mbedtls.dir/ssl_tls.c.o
[ 43%] Linking C static library ../../../library/libmbedtls.a
[ 43%] Built target mbedtls
Scanning dependencies of target sqlite3
[ 43%] Building C object lib/sqlite-amalgamation-3150200/CMakeFiles/sqlite3.dir/sqlite3.c.o
[ 44%] Linking C static library ../../library/libsqlite3.a
[ 44%] Built target sqlite3
Scanning dependencies of target cJSON
[ 44%] Building C object lib/cjson/CMakeFiles/cJSON.dir/src/cJSON.c.o
[ 44%] Linking C static library ../../library/libcJSON.a
[ 44%] Built target cJSON
Scanning dependencies of target co
[ 45%] Building C object lib/flb_libco/CMakeFiles/co.dir/libco.c.o
[ 45%] Linking C static library ../../library/libco.a
[ 45%] Built target co
Scanning dependencies of target flb-plugin-proxy-go
[ 46%] Building C object src/proxy/go/CMakeFiles/flb-plugin-proxy-go.dir/go.c.o
[ 46%] Linking C static library ../../../library/libflb-plugin-proxy-go.a
[ 46%] Built target flb-plugin-proxy-go
Scanning dependencies of target fluent-bit-static
[ 46%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_lib.c.o
[ 47%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_log.c.o
[ 47%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_uri.c.o
[ 47%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_pack.c.o
[ 48%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_sha1.c.o
[ 48%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_pipe.c.o
[ 49%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_kernel.c.o
[ 49%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_input.c.o
[ 50%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_output.c.o
[ 50%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_config.c.o
[ 51%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_network.c.o
[ 51%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_utils.c.o
[ 52%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_engine.c.o
[ 52%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_engine_dispatch.c.o
[ 53%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_task.c.o
[ 53%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_scheduler.c.o
[ 54%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_io.c.o
[ 54%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_upstream.c.o
[ 54%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_router.c.o
[ 55%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_http_client.c.o
[ 55%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_worker.c.o
[ 56%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_io_tls.c.o
[ 56%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_plugin_proxy.c.o
[ 57%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_sqldb.c.o
[ 57%] Linking C static library ../library/libfluent-bit.a
[ 57%] Built target fluent-bit-static
Scanning dependencies of target flb-plugin-out_flowcounter
[ 57%] Building C object plugins/out_flowcounter/CMakeFiles/flb-plugin-out_flowcounter.dir/out_flowcounter.c.o
[ 58%] Linking C static library ../../library/libflb-plugin-out_flowcounter.a
[ 58%] Built target flb-plugin-out_flowcounter
Scanning dependencies of target flb-plugin-out_lib
[ 58%] Building C object plugins/out_lib/CMakeFiles/flb-plugin-out_lib.dir/out_lib.c.o
[ 59%] Linking C static library ../../library/libflb-plugin-out_lib.a
[ 59%] Built target flb-plugin-out_lib
Scanning dependencies of target flb-plugin-out_td
[ 60%] Building C object plugins/out_td/CMakeFiles/flb-plugin-out_td.dir/td_http.c.o
In file included from /root/fluent-bit/plugins/out_td/td_http.c:28:
/root/fluent-bit/plugins/out_td/miniz/miniz.c:1012:14: warning: unused function 'def_realloc_func' [-Wunused-function]
static void *def_realloc_func(void *opaque, void *address, size_t items, size_t size) { (void)opaque, (void)address, (void)items, (void)size; return MZ_REALLOC(address, items * size); }
             ^
1 warning generated.
[ 60%] Building C object plugins/out_td/CMakeFiles/flb-plugin-out_td.dir/td_config.c.o
[ 61%] Building C object plugins/out_td/CMakeFiles/flb-plugin-out_td.dir/td.c.o
[ 61%] Linking C static library ../../library/libflb-plugin-out_td.a
[ 61%] Built target flb-plugin-out_td
Scanning dependencies of target flb-plugin-out_stdout
[ 62%] Building C object plugins/out_stdout/CMakeFiles/flb-plugin-out_stdout.dir/stdout.c.o
[ 62%] Linking C static library ../../library/libflb-plugin-out_stdout.a
[ 62%] Built target flb-plugin-out_stdout
Scanning dependencies of target flb-plugin-out_plot
[ 62%] Building C object plugins/out_plot/CMakeFiles/flb-plugin-out_plot.dir/plot.c.o
/root/fluent-bit/plugins/out_plot/plot.c:163:13: warning: implicit declaration of function 'dprintf' is invalid in C99 [-Wimplicit-function-declaration]
            dprintf(fd, "%lu %" PRIu64 "\n", atime, val->via.u64);
            ^
1 warning generated.
[ 62%] Linking C static library ../../library/libflb-plugin-out_plot.a
[ 62%] Built target flb-plugin-out_plot
Scanning dependencies of target flb-plugin-out_null
[ 62%] Building C object plugins/out_null/CMakeFiles/flb-plugin-out_null.dir/null.c.o
[ 63%] Linking C static library ../../library/libflb-plugin-out_null.a
[ 63%] Built target flb-plugin-out_null
Scanning dependencies of target flb-plugin-out_influxdb
[ 64%] Building C object plugins/out_influxdb/CMakeFiles/flb-plugin-out_influxdb.dir/influxdb_bulk.c.o
[ 64%] Building C object plugins/out_influxdb/CMakeFiles/flb-plugin-out_influxdb.dir/influxdb.c.o
[ 65%] Linking C static library ../../library/libflb-plugin-out_influxdb.a
[ 65%] Built target flb-plugin-out_influxdb
Scanning dependencies of target flb-plugin-out_http
[ 66%] Building C object plugins/out_http/CMakeFiles/flb-plugin-out_http.dir/http.c.o
[ 66%] Linking C static library ../../library/libflb-plugin-out_http.a
[ 66%] Built target flb-plugin-out_http
Scanning dependencies of target flb-plugin-out_forward
[ 66%] Building C object plugins/out_forward/CMakeFiles/flb-plugin-out_forward.dir/__/__/src/flb_network.c.o
[ 67%] Building C object plugins/out_forward/CMakeFiles/flb-plugin-out_forward.dir/forward.c.o
[ 67%] Linking C static library ../../library/libflb-plugin-out_forward.a
[ 67%] Built target flb-plugin-out_forward
Scanning dependencies of target flb-plugin-out_es
[ 67%] Building C object plugins/out_es/CMakeFiles/flb-plugin-out_es.dir/es_bulk.c.o
[ 68%] Building C object plugins/out_es/CMakeFiles/flb-plugin-out_es.dir/es.c.o
[ 68%] Linking C static library ../../library/libflb-plugin-out_es.a
[ 68%] Built target flb-plugin-out_es
Scanning dependencies of target flb-plugin-out_counter
[ 68%] Building C object plugins/out_counter/CMakeFiles/flb-plugin-out_counter.dir/counter.c.o
[ 69%] Linking C static library ../../library/libflb-plugin-out_counter.a
[ 69%] Built target flb-plugin-out_counter
Scanning dependencies of target flb-plugin-in_random
[ 70%] Building C object plugins/in_random/CMakeFiles/flb-plugin-in_random.dir/random.c.o
[ 70%] Linking C static library ../../library/libflb-plugin-in_random.a
[ 70%] Built target flb-plugin-in_random
Scanning dependencies of target flb-plugin-in_forward
[ 71%] Building C object plugins/in_forward/CMakeFiles/flb-plugin-in_forward.dir/fw.c.o
/root/fluent-bit/plugins/in_forward/fw.c:91:41: warning: incompatible pointer types passing 'int (struct flb_config *, void *)' to parameter of type 'int (*)(struct flb_input_instance *, struct flb_config *, void *)' [-Wincompatible-pointer-types]
                                        in_fw_collect,
                                        ^~~~~~~~~~~~~
/root/fluent-bit/include/fluent-bit/flb_input.h:451:42: note: passing argument to parameter 'cb_new_connection' here
                                   int (*cb_new_connection) (struct flb_input_instance *,
                                         ^
/root/fluent-bit/plugins/in_forward/fw.c:124:21: warning: incompatible pointer types initializing 'int (*)(struct flb_input_instance *, struct flb_config *, void *)' with an expression of type 'int (struct flb_config *, void *)' [-Wincompatible-pointer-types]
    .cb_collect   = in_fw_collect,
                    ^~~~~~~~~~~~~
2 warnings generated.
[ 71%] Building C object plugins/in_forward/CMakeFiles/flb-plugin-in_forward.dir/fw_conn.c.o
[ 72%] Building C object plugins/in_forward/CMakeFiles/flb-plugin-in_forward.dir/fw_prot.c.o
[ 72%] Building C object plugins/in_forward/CMakeFiles/flb-plugin-in_forward.dir/fw_config.c.o
[ 73%] Linking C static library ../../library/libflb-plugin-in_forward.a
[ 73%] Built target flb-plugin-in_forward
Scanning dependencies of target flb-plugin-in_lib
[ 74%] Building C object plugins/in_lib/CMakeFiles/flb-plugin-in_lib.dir/in_lib.c.o
[ 74%] Building C object plugins/in_lib/CMakeFiles/flb-plugin-in_lib.dir/__/__/src/flb_pack.c.o
[ 75%] Linking C static library ../../library/libflb-plugin-in_lib.a
[ 75%] Built target flb-plugin-in_lib
Scanning dependencies of target flb-plugin-in_mqtt
[ 75%] Building C object plugins/in_mqtt/CMakeFiles/flb-plugin-in_mqtt.dir/mqtt.c.o
[ 76%] Building C object plugins/in_mqtt/CMakeFiles/flb-plugin-in_mqtt.dir/mqtt_conn.c.o
[ 76%] Building C object plugins/in_mqtt/CMakeFiles/flb-plugin-in_mqtt.dir/mqtt_prot.c.o
[ 77%] Building C object plugins/in_mqtt/CMakeFiles/flb-plugin-in_mqtt.dir/mqtt_config.c.o
[ 77%] Linking C static library ../../library/libflb-plugin-in_mqtt.a
[ 77%] Built target flb-plugin-in_mqtt
Scanning dependencies of target flb-plugin-in_tcp
[ 77%] Building C object plugins/in_tcp/CMakeFiles/flb-plugin-in_tcp.dir/tcp.c.o
[ 78%] Building C object plugins/in_tcp/CMakeFiles/flb-plugin-in_tcp.dir/tcp_conn.c.o
[ 78%] Building C object plugins/in_tcp/CMakeFiles/flb-plugin-in_tcp.dir/tcp_config.c.o
[ 79%] Linking C static library ../../library/libflb-plugin-in_tcp.a
[ 79%] Built target flb-plugin-in_tcp
Scanning dependencies of target flb-plugin-in_stdin
[ 80%] Building C object plugins/in_stdin/CMakeFiles/flb-plugin-in_stdin.dir/in_stdin.c.o
[ 80%] Building C object plugins/in_stdin/CMakeFiles/flb-plugin-in_stdin.dir/__/__/src/flb_pack.c.o
[ 81%] Linking C static library ../../library/libflb-plugin-in_stdin.a
[ 81%] Built target flb-plugin-in_stdin
Scanning dependencies of target flb-plugin-in_serial
[ 82%] Building C object plugins/in_serial/CMakeFiles/flb-plugin-in_serial.dir/in_serial.c.o
[ 82%] Building C object plugins/in_serial/CMakeFiles/flb-plugin-in_serial.dir/in_serial_config.c.o
[ 82%] Linking C static library ../../library/libflb-plugin-in_serial.a
[ 82%] Built target flb-plugin-in_serial
Scanning dependencies of target flb-plugin-in_health
[ 82%] Building C object plugins/in_health/CMakeFiles/flb-plugin-in_health.dir/health.c.o
[ 82%] Linking C static library ../../library/libflb-plugin-in_health.a
[ 82%] Built target flb-plugin-in_health
Scanning dependencies of target flb-plugin-in_head
[ 82%] Building C object plugins/in_head/CMakeFiles/flb-plugin-in_head.dir/in_head.c.o
[ 83%] Linking C static library ../../library/libflb-plugin-in_head.a
[ 83%] Built target flb-plugin-in_head
Scanning dependencies of target fluent-bit-shared
[ 84%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_lib.c.o
[ 84%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_log.c.o
[ 84%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_uri.c.o
[ 85%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_pack.c.o
[ 85%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_sha1.c.o
[ 86%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_pipe.c.o
[ 86%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_kernel.c.o
[ 87%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_input.c.o
[ 87%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_output.c.o
[ 88%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_config.c.o
[ 88%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_network.c.o
[ 89%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_utils.c.o
[ 89%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_engine.c.o
[ 90%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_engine_dispatch.c.o
[ 90%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_task.c.o
[ 90%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_scheduler.c.o
[ 91%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_io.c.o
[ 91%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_upstream.c.o
[ 92%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_router.c.o
[ 92%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_http_client.c.o
[ 93%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_worker.c.o
[ 93%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_io_tls.c.o
[ 94%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_plugin_proxy.c.o
[ 94%] Building C object src/CMakeFiles/fluent-bit-shared.dir/flb_sqldb.c.o
[ 95%] Linking C shared library ../lib/libfluent-bit.so
/usr/bin/ld: cannot find -ldl
cc: error: linker command failed with exit code 1 (use -v to see invocation)
*** Error code 1

Stop.
make[2]: stopped in /root/fluent-bit/build
*** Error code 1

Stop.
make[1]: stopped in /root/fluent-bit/build
*** Error code 1

Stop.
make: stopped in /root/fluent-bit/build
root@FreeBSD-10:~/fluent-bit/build #
```

**Comment by edsiper** [comment]
thanks for reporting this issue.

Since the last versions, our CMake build rules try to link to -ldl using the ${CMAKE_DL_LIBS} variable. Would you please specify which cmake versions your system have ?

```bash
$ cmake --version
```

**Comment by edsiper** [comment]
I've tested on FreeBSD 10 and found the issues.

I have pushed the fixes for [GIT master](http://github.com/fluent/fluent-bit) (incoming 0.11) and in our minor fix branch [0.10](https://github.com/fluent/fluent-bit/tree/0.10).

I will release 0.10.1 this week, so everything should be fine now. Thanks again for reporting this problem!

**Comment by pnovotnak** [comment]
Dude, you rock!
